features/dd-sdk-android-trace/src/main/kotlin/com/datadog/android/trace/_InternalCoreWriterProvider.kt

```diff
@@ -0,0 +1,13 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.trace
+
+import com.datadog.android.api.feature.StorageBackedFeature
+
+interface _InternalCoreWriterProvider : StorageBackedFeature {
+    fun getCoreTracerWriter(): com.datadog.trace.common.writer.Writer
+}
```

Comment:
Can you please describe this one? How it is used, why does it start with `_` and extends `StorageBackedFeature`? We are mixing here feature and provider concept. This one, probably, can be a standalone interface, without extending `StorageBackedFeature`.

Reply:
This is being used here: https://github.com/DataDog/dd-sdk-android/pull/2021/files#diff-1db0ecd609920bfa147ebcdc3403f1534b1d66e6506d8a055297399552783c5fR140 

As you can see it is outside of the module and `unwrap` function only accepts a `<T:Feature>` , therefor the need for that interface to extend `StorageBackedFeature`.
The reason why I added `_` as a prefix is to follow the `_InternalProxy` approach for RN and Flutter to expose that class but mark it as Internal.
Please let me know if you have a better idea to export that `coreTracerWriter` from there in the `dd-sdk-android-trace-otel` module.

Reply:
I still think it is better to keep them separate and don't use `_` prefix. `Feature` interface is indented for communication with SDK core, and here we are adding something completely not related.

IMO Instead, it is better to have just `CoreWriterProvider` interface, not extending any interface, annotated with `@InternalApi`, which has only one method `getCoreTracerWriter`.

And then in the code:

```
val tracingFeature = sdkCore.getFeature(Feature.TRACING_FEATURE_NAME).unwrap()
...
val coreTracerWriter = (tracingFeature as? `CoreWriterProvider`).getCoreTracerWriter()
```